components/annot-vocabulary.vue

```diff
@@ -1,299 +1,356 @@
+                // Bootstrap variant color of the button to save the annotation to the data table
+                return ( !this.saveButtonDisabled ) ? "success" : "secondary";
+            }
+        },
+
+        mounted() {
+            
+            // Initialize the mapping of all unique values as null
+            this.initializeMapping();
+        },        
+
+        methods: {
+
+            applyAnnotation() {
+
+                // If this is 'row' mode
+                if ( "row" === this.options.mode ) {
+
+                    // We want to use the annotated dataTable here in order to not overwrite previous
+                    // annotations from other components
+
+                    // 1. Create a local copy of the annotated table for transformation
+                    const transformedTable = structuredClone(this.dataTable.annotated);
+
+                    // 2. Transform all values in columns categorized as 'age' columns
+                    for ( let index = 0; index < transformedTable.length; index++ ) {
+                        for ( const columnName in transformedTable[index] ) {
+
+                                if ( this.relevantColumns.includes(columnName) ) {
+
+                                    // TODO: if "value" is a missing value or doesn't fit the heuristic, this will currently break!
+                                    transformedTable[index][columnName] = this.transformedValue(columnName, transformedTable[index][columnName]);
+                                }
+                        }
+                    }
+
+                    // 3. Trigger a save of this transformation to the annotated table in the store
+                    this.$emit("update:dataTable", {
+
+                        transformHeuristics: this.vocabularyMapping,
+                        transformedTable: transformedTable
+                    });
+                } 
+                // Else, this is 'column' mode
+                else {
+
+                    this.$emit("update:heuristics", {
+
+                        transformHeuristics: this.vocabularyMapping,
+                    });
+                }
+            },          
+
+            checkAnnotationState() {
+
+                let columnHasUnmappedValues = false;
+
+                if ( "row" === this.options.mode ) {
+
+                    // 1. Look for unannotated values in the vocabulary map
+                    for ( const columnName in this.vocabularyMapping ) {
+
+                        // A. Check to see if there is at least one unannotated value for this column
+                        if ( Object.values(this.vocabularyMapping[columnName])
+                                   .some((uniqueValue) => null === uniqueValue) ) {
+
+                            columnHasUnmappedValues = true;
+                            break;
+                        }
+                    }
+                } 
+                // Else, this is in "column" mode
+                else {
```

Comment:
as in the above, I would be explicit here. The fact that we add a comment is a good indicator that the code could use it